Assets/_GameJamCat/Scripts/PlayerController/PlayerController.cs

```diff
@@ -0,0 +1,196 @@
+
+            //Managing Player States
+            if (characterController.isGrounded)
+            {
+                if (hInput == 0 && vInput == 0)
+                    playerStates = PlayerStates.Idle;
+                else
+                {
+                    if (_speed == walkSpeed)
+                        playerStates = PlayerStates.Walking;
+                    else
+                        playerStates = PlayerStates.Running;
+
+                    _footstepDelay = (2/_speed);
+                }
+            }
+            else
+                playerStates = PlayerStates.Jumping;
+        }
+
+        void Jump()
+        {
+            if (Input.GetButtonDown(JumpInput))
+            {
+                StartCoroutine(PerformJumpRoutine());
+                JumpAnimation = true;
+            }
+        }
+
+        IEnumerator PerformJumpRoutine()
+        {
+            //play jump sound
+            
+
+            float _jump = JumpForce;
+
+            do
+            {
+                characterController.Move(Vector3.up * _jump * Time.deltaTime);
+                _jump -= Time.deltaTime;
+                yield return null;
+            }
+            while (!characterController.isGrounded);
+
+
+        }
+
+        void SetCharacterAnimations()
+        {
+            if (!CharacterAnimator)
+                return;
+
+            switch (playerStates)
+            {
+                case PlayerStates.Idle:
+                    HorzAnimation = Mathf.Lerp(HorzAnimation, 0, 5 * Time.deltaTime);
+                    VertAnimation = Mathf.Lerp(VertAnimation, 0, 5 * Time.deltaTime);
+                    break;
+
+                case PlayerStates.Walking:
+                    HorzAnimation = Mathf.Lerp(HorzAnimation, 1 * Input.GetAxis("Horizontal"), 5 * Time.deltaTime);
+                    VertAnimation = Mathf.Lerp(VertAnimation, 1 * Input.GetAxis("Vertical"), 5 * Time.deltaTime);
+                    break;
+
+                case PlayerStates.Running:
+                    HorzAnimation = Mathf.Lerp(HorzAnimation, 2 * Input.GetAxis("Horizontal"), 5 * Time.deltaTime);
+                    VertAnimation = Mathf.Lerp(VertAnimation, 2 * Input.GetAxis("Vertical"), 5 * Time.deltaTime);
+                    break;
+
+                case PlayerStates.Jumping:
+                    if (JumpAnimation)
+                    {
+                        CharacterAnimator.SetTrigger("Jump");
+                        JumpAnimation = false;
+                    }
+                    break;
+            }
+
+            LandAnimation = characterController.isGrounded;
+            CharacterAnimator.SetFloat("Horizontal", HorzAnimation);
+            CharacterAnimator.SetFloat("Vertical", VertAnimation);
+            CharacterAnimator.SetBool("isGrounded", LandAnimation);
+        }
+
+        bool onGround()
+        {
+            bool retVal = false;
+
+            if (Physics.Raycast(FootLocation.position, Vector3.down, 0.1f))
```

Comment:
just do 
return Physics.Raycast(FootLocation.position, Vector3.down, 0.1f) 
for the whole method